### PR TITLE
player and shroom

### PR DESCRIPTION
**Pull request by RSGameDev**
work in progress.
hope its enough for a push back feature to be made from.
rook has not been completed here.

### CODE REVIEW

**Comment by RSGameDev** [comment]
I've pushed this but realise the code style may need fixing by me. Like how I name variables uppercase lowercase etc. I can do this in the near future. I just wanted to get this on the repo asap just so u have it.

File: bpdv3/Assets/Scripts/Floor/TileController.cs [hunk 1]
```diff
@@ -19,7 +19,7 @@ public class TileController : MonoBehaviour {
     Vector3 direction;
     public float distance;
 
-    private bool _isMoving = false;
+    public bool _isMoving = false;
```
Comment (KEREMADALI) [inline bpdv3/Assets/Scripts/Floor/TileController.cs]: update name according to accessor

File: bpdv3/Assets/Scripts/Player/PlayerDestination.cs [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            IsObtained = true;
+        }
+    }
+
+    private void Update()
+    {
+        PositionContraint(VacantDestination);
+    }
+
+    private void PositionContraint(Collider other)
```
Comment (KEREMADALI) [inline bpdv3/Assets/Scripts/Player/PlayerDestination.cs]: Please define a self- explanatory name

File: bpdv3/Assets/Scripts/Player/PlayerMovement.cs [hunk 3]
```diff
@@ -2,109 +2,125 @@
...
-            onBeat = hitZone.GetComponent<CoreHitzone>().onBeat;
-            direction = "w";
-            transform.LookAt(transform.position + Vector3.forward);
-            transform.position += new Vector3(0, 0, 1.25f);
-            if (transform.position.z > 7)
+            if (Input.GetKeyDown(KeyCode.W))
+            {
+                IsInput = true;
+                PlayerDestination.IsObtained = false;
+                /////IsOnBeat = BeatHitZone.GetComponent<CoreHitzone>().onBeat;
```
Comment (KEREMADALI) [inline bpdv3/Assets/Scripts/Player/PlayerMovement.cs]: zombie codes can be deleted

File: bpdv3/Assets/Scripts/enemy/EnemyNextMove.cs [hunk 4]
```diff
@@ -5,31 +5,136 @@
...
+            //    case Enemy.EnemyType.Shroom:
+                    if (!IsDestinationObtained)
+                    {
+                        IsDestinationObtained = true;
+                        EnemyMovement.NextMoveLocationGO = other.gameObject;
+                        OccupiedNumber = other.GetComponentInParent<TileProperties>().occupiedNum;
+                        gameObject.GetComponent<Collider>().enabled = false;
+                    }
+                    //break;
+                //case Enemy.EnemyType.Rook:
```
Comment (KEREMADALI) [inline bpdv3/Assets/Scripts/enemy/EnemyNextMove.cs]: Please remove if the code is not necessary

File: bpdv3/Assets/Scripts/enemy/EnemyNextMove.cs [hunk 5]
```diff
@@ -5,31 +5,136 @@
...
+        else if (other.gameObject.layer == LayerMask.NameToLayer(s_noGoMoveLayer))
+        {
+            OccupiedNumber = other.GetComponentInParent<TileProperties>().occupiedNum;
+            other.gameObject.GetComponentInParent<TileProperties>().OccupiedDecreased();
+            gameObject.GetComponent<Collider>().enabled = false;
+            IsOutOfBounds = true;
+        }       
     }
 
+    //private void OnTriggerEnter(Collider other)
```
Comment (KEREMADALI) [inline bpdv3/Assets/Scripts/enemy/EnemyNextMove.cs]: Same

**Review by KEREMADALI — COMMENTED** [review]
Clean coding, thanks.

**Pull request merged**